proxy/proxy.go

```diff
@@ -590,11 +591,21 @@ func (proxy *Proxy) updateContainerNetworkSettings(container jsonObject) error {
 		return err
 	}
 
+	state, err := container.Object("State")
+	if err != nil {
+		return err
+	}
+
+	pid, err := state.Int("Pid")
+	if err != nil {
+		return err
+	}
+
 	if err := proxy.waitForStartByIdent(containerID); err != nil {
 		return err
 	}
-	mac, ips, nets, err := weaveContainerIPs(containerID)
-	if err != nil || len(ips) == 0 {
+	weaveInterfaces, err := GetWeaveInterfaces(proxy.ProcPath, pid)
```

Comment:
you could have another variant of this that searched for exactly "ethwe", so you wouldn't have to arbitrarily pick the first answer.